src/main/php/util/data/Sequence.class.php

```diff
@@ -42,18 +42,21 @@ protected function __construct($elements) {
    *
    * @param  function(): var $operation
    * @return var
+   * @throws util.data.CannotReset
    */
   protected function terminal($operation) {
     static $message= 'Underlying value is streamed and cannot be processed more than once';
 
     try {
       return $operation();
-    } catch (IllegalStateException $e) {
-      throw new IllegalStateException($message, $e);
+    } catch (CannotReset $e) {
+      throw new CannotReset($message, $e);
     } catch (Throwable $e) {
       throw $e;
-    } catch (\Exception $e) {
-      throw new IllegalStateException($message.':'.$e->getMessage());
+    } catch (\Throwable $e) {   // PHP7
+      throw new CannotReset($message.':'.$e->getMessage());
+    } catch (\Exception $e) {   // PHP5
+      throw new CannotReset($message.':'.$e->getMessage());
     }
```

Comment:
As this now catches everything and the `SequenceIterator` does as well, can't the `iterator()` method in line 64 be simplified to just return the `SequenceIterator` directly?